docs/data-structures.md

```diff
@@ -0,0 +1,249 @@
+| `args`        | [Bytes] | **An array of arguments for the script.** The arguments here are imported into the CKB-VM instance as input arguments for the scripts. This part is NOT used when calculating the hash of the script. |
+| `signed_args` | [Bytes] | **An array of signed arguments for the script.** The arguments with signatures. The `signed_args` and the `args` will be connected into a single vector, and imported into the CKB-VM instance as input arguments. |
+
+
+
+More information about Script can be [here](https://github.com/nervosnetwork/ckb-demo-ruby-sdk/blob/27669cd6b4f56f8977b725fc0e6582b288ab2b82/docs/how-to-write-contracts.md#script-model).
+
+
+
+## Transaction
+
+### Example
+
+```json
+{
+    "deps": [],
+    "hash": "0x4707810253259258e3091934dc5b543403c10cc899859e077fe26067f8d52dc0",
+    "inputs": [
+      {
+        "previous_output": {
+          "hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+          "index": 4294967295
+        },
+        "unlock": {
+          "args": [],
+          "binary": "0x0d00000000000000",
+          "reference": null,
+          "signed_args": [],
+          "version": 0
+        }
+      }
+    ],
+    "outputs": [
+      {
+        "capacity": 5000000,
+        "data": "0x",
+        "lock": "0xa58a960b28d6e283546e38740e80142da94f88e88d5114d8dc91312b8da4765a",
+        "type": null
+      }
+    ],
+    "version": 0
+}
+```
+
+### Description
+
+#### Transaction
+
+| Name              | Type                             | Description                                                  |
+| ----------------- | -------------------------------- | ------------------------------------------------------------ |
+| `version`         | uint32                           | **The version of the transaction.** It‘s used to distinguish transactions when there's a fork happened to the blockchain system. |
+| `deps`            | [`outpoint`]                     | **An array of `outpoint` that point to the cells that are dependencies of this transaction.** Only live cells can be listed here. The cells listed are read-only. |
+| `inputs`          | [{`previsou_output` , `unlock`}] | **An array of {`previsou_output` , `unlock`}.**              |
+| `previous_output` | `outpoint`                       | **A cell outpoint that point to the cells used as inputs.** Input cells are in fact the output of previous transactions, hence they are noted as `previous_output` here. These cells are referred through  `outpoint`, which contains the transaction `hash` of the previous transaction, as well as this cell's `index` in its transaction's output list. |
+| `unlock`          | `script`                         | **A script for unlocking the corresponding input cell** (i.e. `previous_output`). See [here](https://github.com/nervosnetwork/ckb-demo-ruby-sdk/blob/develop/docs/how-to-write-contracts.md) for how to program this part. |
```

Comment:
The `previous_output` and `unlock` should not be included, they are part of `inputs`, not the transaction.

Reply:
The fact that `inputs` is part of transaction and `previsou_output` and `unlock` is part of `input` makes them a part of `inputs` as well.

A keyboard is a part of a laptop, which does not make any of the keys is not part of it.

So I don't think this part could cause too much confusion.

More importantly, this docs serve as a dictionary for looking up parameters, its more convenient  to put related things near each other.